api/insight/apimiddleware.go

```diff
@@ -52,6 +60,16 @@ func (c *insightApiContext) getBlockHashCtx(r *http.Request) string {
 	return hash
 }
 
+// AddressPathCtx returns a http.HandlerFunc that embeds the value at the url
+// part {address} into the request context.
+func (c *insightApiContext) AddressPathCtx(next http.Handler) http.Handler {
```

Comment:
Thank you for the review.  Whenever I tweaked or threatened to tweak one of the middleware functions I just duplicated it to avoid possible bugs introduced into the other API.   Not a problem though, I just removed it and switched back to the m. context one.

Reply:
One more issue.  ctxAddress is also set with with:
```go
func (c *insightApiContext) PostAddrsTxsCtx(next http.Handler) http.Handler {
```
So it looks like I have to find a way to export ctxAddress from the other context into the insight one.

Reply:
I might be missing the issue, but I think you would use `PostAddrsTxsCtx` from middleware instead of insight

Reply:
oh, nvm, middleware has `AddressPathCtx` and `GetAddressCtx`, but not `PostAddrsTxsCtx`.  I see...

Reply:
How about changing `PostAddrsTxsCtx` to set `m.CtxAddress` instead of `ctxAddress`.  Then you don't need an insight version of AddressPathCtx and GetAddressCtx

Reply:
Worked perfectly!   BTW, we could probably get rid of `GetRawHexTx` this same way if you want (different PR?)